libsolidity/interface/CompilerStack.h

```diff
@@ -133,12 +133,54 @@ class CompilerStack: public langutil::CharStreamProvider, public evmasm::Abstrac
 		SolidityAST,
 	};
 
-	enum class IROutputSelection {
-		None,
-		UnoptimizedOnly,
-		UnoptimizedAndOptimized,
+	/// Indicates which stages of the compilation pipeline were explicitly requested and provides
+	/// logic to determine which ones are effectively needed to accomplish that.
+	/// Note that parsing and analysis are not selectable, since they cannot be skipped.
+	struct PipelineSelection
+	{
+		bool irCodegen = false;      ///< Want IR output straight from code generator.
+		bool irOptimization = false; ///< Want reparsed IR that went through YulStack. May be optimized or not, depending on settings.
+		bool bytecode = false;       ///< Want EVM-level outputs, especially EVM assembly and bytecode. May be optimized or not, depending on settings.
```

Comment:
I had a really hard time coming up with good names for this, and I'm still not fully satisfied.

Initially I went with `OutputSelection`, but this does not really apply to all the outputs, just those that require going into new branches of he pipeline. `StandardCompiler` (and Standard JSON) already have a thing called "output selection" and it's not exactly this so that naming would be confusing.

Same for the fields - e.g. `bytecode` could mean just generating EVM assembly so it's not perfect. I considered naming this `evm` or `evmasm`, but `bytecode` seems like a better overall name, even if it's not precise.

Reply:
Perhaps `PipelineComponentSelection`? Or `PipelineStageSelection`? That would at least make it clearer in terms of not selecting a whole different pipeline, just branches of it.

Reply:
It does sound a bit ambiguous, because I didn't want to explicitly call it a "stage" and because to some extent it does select a pipeline (when not compiling via IR, selecting `--ir` really enables the second pipeline).

"Stage" does not sound right because it implies sequentially going through stages like parsing/analysis/codegen/optimization/assembling/linking/etc. and this is not exactly what we are selecting here. `CompilerStack::State`/`m_stopAfter` covers exactly this progression and I didn't want confusion with those. What the new struct is describing is more how to putting together a potentially branching franken-pipeline from parts lying around :) I would not really call these "components" though.

I decided to rename it to `PipelineConfig` now to remove the ambiguity, but it's still not that great IMO.

Reply:
Definitely are - thanks for the elaboration! I'm fine with it in any case